infrastructure/k8s/readme.md

```diff
@@ -0,0 +1,290 @@
+```sh
+helm repo add jetstack https://charts.jetstack.io
+```
+
+Update your local Helm chart repository cache
+```sh
+helm repo update
+```
+
+Install the cert-manager Helm chart to enable support for lets-encrypt.
+```sh
+helm install cert-manager --namespace cert-manager --version v0.11.0 jetstack/cert-manager
+```
+
+## Monitoring
+To use Azure insights we need an analytics workspace to send data into. 
+```sh
+az resource create --resource-type Microsoft.OperationalInsights/workspaces \
+ --name coverServiceDDBWorkspace \
+ --resource-group ${res} \
+ --location ${region} \
+ --properties '{}' -o table
+```
+ 
+First get the resource id of the workspace you created, by running
+```sh
+workspaceresoid=$(az resource show --resource-type Microsoft.OperationalInsights/workspaces --resource-group ${res} --name coverServiceDDBWorkspace --query "id" -o tsv)
+```
+
+Next enable the monitoring add-on by running the command below.
+```sh
+az aks enable-addons --resource-group ${res} \
+    --name ${ksname} \
+    --addons monitoring \
+    --workspace-resource-id ${workspaceresoid}
+```
+
+
+Send pod container logs to insights and enabled live view.
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+    name: containerHealth-log-reader
+rules:
+    - apiGroups: ["", "metrics.k8s.io", "extensions", "apps"]
+      resources:
+         - "pods/log"
+         - "events"
+         - "nodes"
+         - "pods"
+         - "deployments"
+         - "replicasets"
+      verbs: ["get", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+    name: containerHealth-read-logs-global
+roleRef:
+    kind: ClusterRole
+    name: containerHealth-log-reader
+    apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: User
+  name: clusterUser
+  apiGroup: rbac.authorization.k8s.io
+```
+
+```sh
+kubectl apply -f logreader-rbac.yaml
+```
+
+## Horizontal pod autoscaler (HPA)
+
+The application deployment use HPA see the app deployment for more information.
+
+# Install elastic search operator
+
+```sh
+kubectl apply -f https://download.elastic.co/downloads/eck/1.0.0-beta1/all-in-one.yaml
+```
+
+Use this after you have installed the elastic-search application.
+```sh
+kubectl get secret elasticsearch-es-elastic-user -o=jsonpath='{.data.elastic}' | base64 --decode
+```
+
+# Github package registry
+To use the github package registry from kubernetes you need to store basic authentication in a secret and use it in `imagePullSecrets` in the yaml deployment.
+
+```sh
+kubectl create secret docker-registry github-registry \
+    --docker-server=docker.pkg.github.com \
+    --docker-username=cableman \
+    --docker-password=XXXXX_TOKEN_XXXXX \
+    --docker-email=jeskr@aarhus.dk
+```
+
+See the github secret in clear text from the cluster.
+```sh
+kubectl get secret github-registry --output="jsonpath={.data.\.dockerconfigjson}" | base64 --decode
+```
+
+# Application install
```

Comment:
Missing comments about what is going on...

Reply:
Fixed